File: 134-Gas-Station/134-Gas-Station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int totalg = 0;
        int currg = 0;
        int startInd = 0;
        for(int i = 0; i< gas.size(); i++){
            currg += gas[i] - cost[i];
            totalg += gas[i] - cost[i];
            if(currg < 0){
                startInd = i+1;
                currg = 0;
            }
        }
        return totalg >= 0 ? startInd : -1;
    }
};

File: 134-Gas-Station/134-Gas-Station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        totalg, currg, startInd = 0, 0, 0
        for i in range(0, len(gas)):
            currg += gas[i] - cost[i]
            totalg += gas[i] - cost[i]
            if currg < 0:
                startInd = i + 1
                currg = 0
        return startInd if totalg >= 0 else -1
